integrations/spark/spark-job-dataquality/spark-job-dataquality-app/src/main/java/com/thinkbiganalytics/spark/dataquality/output/DataQualityWriter.java

```diff
@@ -0,0 +1,192 @@
+    }
+
+    /**
+     * Write result to Hive table
+     *
+     * @param scs SparkContextService
+     * @param outputTable Name of the destination Hive table
+     * @param partitionValue Value of the partition for Hive. Currently the processing_dttm column
+     * @return boolean indicating result of write
+     */
+    public boolean writeResultToTable(SparkContextService scs, String outputTable, String partitionValue) {
+
+        boolean retVal = false;
+
+        if (sc == null) {
+            log.error("Error writing result: Spark context is not available.");
+        } else if (hiveContext == null) {
+            log.error("Error writing result: Hive context is not available.");
+        } else if (dqRows.isEmpty()) {
+            log.info("Error writing results: No data to write");
+        } else {
+
+            JavaRDD<DataQualityRow> dqRowsRDD = sc.parallelize(dqRows);
+            DataSet dqRowsDF = scs.toDataSet(hiveContext.createDataFrame(dqRowsRDD, DataQualityRow.class));
+
+            String tempTable = "DQ_DF_TABLE";
+            dqRowsDF.registerTempTable(tempTable);
+
+            createOutputTableIfNotExists(scs, outputTable);
+            writeResultToOutputTable(scs, tempTable, outputTable, partitionValue);
+
+            retVal = true;
+        }
+
+        return retVal;
+    }
+
+    /**
+     * Creates target Hive table if it does not already exist
+     * 
+     * @param scs SparkContextService
+     * @param outputTable Name of the destination Hive table
+     */
+    private void createOutputTableIfNotExists(SparkContextService scs, String outputTable) {
+        String createTableSQL = "CREATE TABLE IF NOT EXISTS " + outputTable
+                                + "\n"
+                                + "(rule_name STRING, rule_description STRING, status BOOLEAN, rule_resultdetail STRING)\n"
+                                + "PARTITIONED BY ( processing_dttm STRING)\n";
+
+        log.info("Executing: " + createTableSQL);
+        scs.sql(hiveContext, createTableSQL);
+    }
+
+    /**
+     * Writes the output to the destination Hive table
+     * 
+     * @param scs SparkContextService
+     * @param tempTable Name of the temp table in Spark
+     * @param outputTable Name of the destination Hive table
+     * @param partitionValue Value of the partition for Hive. Currently the processing_dttm column
+     */
+    private void writeResultToOutputTable(SparkContextService scs, String tempTable, String outputTable, String partitionValue) {
+        String insertTableSQL = "INSERT OVERWRITE TABLE " + outputTable
+                                + " PARTITION ( processing_dttm = "
+                                + partitionValue
+                                + ")"
+                                + " SELECT ruleName, description, status, resultDetail"
+                                + " FROM "
+                                + tempTable;
+
+        log.info("Executing: " + insertTableSQL);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the built-in formatting to construct this argument. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2629)